### PR TITLE
add rule image-src-attribute-not-empty

### PR DESCRIPTION
**Pull request by elisecodedestrucs**
add a rules that forbids the use of  without or with an empty src attribute to save CPU usage.

### CODE REVIEW

**Review by dedece35 — DISMISSED** [review]

File: eslint-plugin/docs/rules/image-src-attribute-not-empty.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# Disallow usage of image with empty source attribute (`@ecocode/image-src-attribute-not-empty`)
```
Comment (utarwyn) [inline eslint-plugin/docs/rules/image-src-attribute-not-empty.md:1]: I suggest to follow conventions and start the rule name by `no-`. But it also impacts name of your files..
```suggestion
# Disallow usage of image with empty source attribute (`@ecocode/no-empty-image-src-attribute`)
```

File: eslint-plugin/tests/lib/rules/image-src-attribute-not-empty.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+};
+const expectedError2 = {
+  messageId: "SpecifySrcAttribute",
+  type: "JSXOpeningElement",
+};
+
+ruleTester.run("image-src-attribute-not-empty", rule, {
+  valid: [
+    `
+      <img src='logo.svg' alt='Ceci est une image svg'/>
```
Comment (utarwyn) [inline eslint-plugin/tests/lib/rules/image-src-attribute-not-empty.js:51]: Can you translate all alternative attributes in this file?
```suggestion
      <img src='logo.svg' alt='This is a SVG image'/>
```

File: eslint-plugin/docs/rules/image-src-attribute-not-empty.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+⚠️ This rule _warns_ in the ✅ `recommended` config.
+
+<!-- end auto-generated rule header -->
+
+## Rule Details
+
+This rule aims to prohibit the usage of <img/> without specifying its source attribute because it causes extra and unnecessary http requests.
```
Comment (utarwyn) [inline eslint-plugin/docs/rules/image-src-attribute-not-empty.md:9]: Can you explain here that this rule is for React/JSX?

**Review by utarwyn — CHANGES_REQUESTED** [review]
Thank you for this contribution too! Also few comments on this one..
Can you also run the linter on added files to fix the build?

**Comment by elisecodedestrucs** [comment]
> Thank you for this contribution too! Also few comments on this one.. Can you also run the linter on added files to fix the build?

When I run the linter it raises errors on files I haven't changed, I don't know what to make of it?

**Comment by utarwyn** [comment]
> @elisecodedestrucs 
> When I run the linter it raises errors on files I haven't changed, I don't know what to make of it?

You can install the prettier plugin in your IDE to format automatically files when you save them ([more info here](https://prettier.io/docs/en/editors.html)).
Otherwise, you can see the errors raised by the build script directly from GitHub [here](https://github.com/green-code-initiative/ecoCode-javascript/pull/19/files#diff-b964b8456c55f83c6988ba5e435eaee6ebfd70d0a5429d2111f28165b872e2b7R39)

**Review by utarwyn — APPROVED** [review]
LGTM :100: Can you also propose a test case for this rule in the test project?

**Review by utarwyn — CHANGES_REQUESTED** [review]
Can you please correct conflicts? (due to the merge of #18)

**Review by utarwyn — APPROVED** [review]

**Pull request merged**